halinuxcompanion/dbus.py

```diff
@@ -61,10 +62,13 @@
 }
 
 
-async def get_interface(bus, service, path, interface) -> ProxyInterface:
-    introspection = await bus.introspect(service, path)
-    proxy = bus.get_proxy_object(service, path, introspection)
-    return proxy.get_interface(interface)
+async def get_interface(bus, service, path, interface) -> Union[ProxyInterface, None]:
```

Comment:
nit: I've seen this `Union[T, None]` pattern in other projects recently (I think HASS), but curious: why not just `Optional[T]`?

Reply:
Mostly because this project also uses it:
https://github.com/muniter/halinuxcompanion/blob/dc6a827fb8778a51730b5ae6e9b6a3283a8b55cd/halinuxcompanion/api.py#L12

I'll gladly change to `Optional` (which is what `mypy` will use in the error output anyway).

Reply:
Agree using optional is better!

Reply:
Perfect! Updated the code to use it.